Composer/packages/client/src/CreationFlow/index.js

```diff
@@ -21,8 +21,27 @@ export function CreationFlow(props) {
   const [step, setStep] = useState();
   // eslint-disable-next-line react/prop-types
   const { creationFlowStatus, setCreationFlowStatus } = props;
-  const { fetchTemplates, openBotProject, createProject, saveProjectAs, saveTemplateId, fetchStorages } = actions;
-  const { templateId, templateProjects, focusedStorageFolder } = state;
+  const {
+    fetchTemplates,
+    openBotProject,
+    createProject,
+    saveProjectAs,
+    saveTemplateId,
+    fetchStorages,
+    fetchFolderItemsByPath,
+  } = actions;
+  const { templateId, templateProjects, focusedStorageFolder, storages } = state;
+  const currentStorageIndex = useRef(0);
+  const currentStorageId = storages[currentStorageIndex.current] ? storages[currentStorageIndex.current].id : 'default';
```

Comment:
Cache `storages[currentStorageIndex.current]` so we don't need to get it twice.

In JavaScript, every member accessor is very expensive, unlike C#. This is because JavaScript is prototypical object-oriented model, there are no virtual tables. In contrast, C# is inheritance OO model. So when writing code, try to reduce number of `.` (dots), `[]` (indexer), etc.